src/components/api/devfile-registry.factory.ts

```diff
@@ -144,4 +140,51 @@ export class DevfileRegistry {
 
     return parseInt(memoryLimitNumber, 10) * Math.pow(1024, power);
   }
+
+  private devfileYamlToJson(yaml: string): che.IWorkspaceDevfile {
+    try {
+      return jsyaml.load(yaml);
+    } catch (e) { }
+  }
+
+  private _fetchDevfiles(location: string): ng.IPromise<Array<IDevfileMetaData>> {
+    let promise = this.$http({
+      'method': 'GET',
+      'url': `${location}/devfiles/index.json`,
+      'headers': this.headers
+    });
+
+    return promise.then((result: any) => {
+      return result.data
+        .map(devfile => {
+          // set the origin url as the location
+          devfile.location = location;
+          return devfile;
+        })
+        .map(devfile => this.updateIconUrl(devfile, location))
+        .map(devfile => this.updateMemoryLimit(devfile));
+    });
+  }
+
+  private updateIconUrl(devfile: IDevfileMetaData, url: string): IDevfileMetaData {
+    if (devfile.icon && !devfile.icon.startsWith('http')) {
+      devfile.icon = url + devfile.icon;
```

Comment:
Could you use ```devfile.icon =  new URL(devfile.icon, url); ``` instead for covering all possible cases.
WDYT?

Reply:
do you mean here to replace `devfile.icon = url + devfile.icon;` with `devfile.icon = new URL(devfile.icon, url);` ? or something else?

Reply:
Yes. I mean just replacing:
``` devfile.icon = url + devfile.icon; ```    ->   ```devfile.icon = new URL(devfile.icon, url).href;```